### PR TITLE
Introduce CharReaderBuilder::ecma404Mode

### PR DESCRIPTION
**Pull request by axic**
Instead of a finished PR, this is an initial one to judge appetite for it.

I found out through testing that our assumption what strict means doesn't exactly match. Strictness according to ECMA-404 is a bit closer what we expected. For reference: [ECMA-404](https://www.ecma-international.org/wp-content/uploads/ECMA-404_2nd_edition_december_2017.pdf) and [ECMA-262](https://tc39.es/ecma262/multipage/structured-data.html#sec-json.parse). Additionally [RFC4627](https://datatracker.ietf.org/doc/html/rfc4627#section-3) implies no BOM is needed.

If this feature is welcome, I can look into adding tests for it.

### CODE REVIEW

**Review by cdunn2001 — APPROVED** [review]
Please rebase.

I would suggest bumping the minor version number, but I doubt this will be a commonly used feature anyway. Bumping the micro number (and soversion) should suffice. We'll include this is the next release, probably `1.9.6`.

**Comment by axic** [comment]
@cdunn2001 done!

File: meson.build [hunk 1]
```diff
@@ -9,7 +9,7 @@ project(
   # 2. /include/json/version.h
   # 3. /CMakeLists.txt
   # IMPORTANT: also update the SOVERSION!!
-  version : '1.9.4',
```
Comment (axic) [inline meson.build:12]: This was missed in c4904b2c0d461f84fba88a760161212651e4f536

**Comment by axic** [comment]
@cdunn2001 let me know if there's anything else I need to do here?

**Comment by axic** [comment]
@cdunn2001 @baylesj @BillyDonahue is anything else needed here?

**Comment by axic** [comment]
Any chance to get this merged?

**Comment by baylesj** [comment]
> Any chance to get this merged?

Thanks for waiting what turns out to have been an incredibly long time. Finally merging this now that I have fixed the CI system.

**Pull request merged**